src/OpenTelemetry.AutoInstrumentation/Instrumentations/NLog/Bridge/OpenTelemetryNLogConverter.cs

```diff
@@ -0,0 +1,199 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Reflection;
+using OpenTelemetry.AutoInstrumentation.DuckTyping;
+using OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.TraceContextInjection;
+#if NET
+using OpenTelemetry.AutoInstrumentation.Logger;
+#endif
+using OpenTelemetry.AutoInstrumentation.Logging;
+using OpenTelemetry.Logs;
+using Exception = System.Exception;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.Bridge;
+
+/// <summary>
+/// Converts NLog LogEventInfo into OpenTelemetry LogRecords.
+/// </summary>
+internal class OpenTelemetryNLogConverter
+{
+    private const int TraceOrdinal = 0;
+    private const int DebugOrdinal = 1;
+    private const int InfoOrdinal = 2;
+    private const int WarnOrdinal = 3;
+    private const int ErrorOrdinal = 4;
+    private const int FatalOrdinal = 5;
+    private const int OffOrdinal = 6;
+
+    private static readonly IOtelLogger Logger = OtelLogging.GetLogger();
+    private static readonly Lazy<OpenTelemetryNLogConverter> InstanceField = new(InitializeTarget, true);
+
+    private readonly Func<string?, object?>? _getLoggerFactory;
+    private readonly ConcurrentDictionary<string, object> _loggers = new(StringComparer.Ordinal);
+
+#if NET
+    private int _warningLogged;
+#endif
+
+    private OpenTelemetryNLogConverter(LoggerProvider loggerProvider)
+    {
+        _getLoggerFactory = CreateGetLoggerDelegate(loggerProvider);
+    }
+
+    public static OpenTelemetryNLogConverter Instance => InstanceField.Value;
+
+    [DuckReverseMethod]
+    public string Name { get; set; } = nameof(OpenTelemetryNLogConverter);
+
+    [DuckReverseMethod(ParameterTypeNames = new[] { "NLog.LogEventInfo, NLog" })]
+    public void WriteLogEvent(ILoggingEvent loggingEvent)
```

Comment:
When is this WriteLogEvent being called? It seems like all of the calls go through OpenTelemetryTarget unless I'm missing something.